src/Lean/Meta/ExprDefEq.lean

```diff
@@ -888,14 +888,15 @@ def checkAssignment (mvarId : MVarId) (fvars : Array Expr) (v : Expr) : MetaM (O
       let v ← instantiateMVars v
       CheckAssignment.checkAssignmentAux mvarId fvars hasCtxLocals v
 
-private def processAssignmentFOApproxAux (mvar : Expr) (args : Array Expr) (v : Expr) : MetaM Bool :=
+private partial def processAssignmentFOApproxAux (mvar : Expr) (args : Array Expr) (v : Expr) : MetaM Bool :=
   match v with
+  | .mdata _ e   => processAssignmentFOApproxAux mvar args v
```

Comment:
There is typo here. `v` => `e`

Reply:
We can also fix it using 
```lean
match v.cleanupAnnotations with
```

Reply:
Ahh, that's what I get for not rechecking the test case after refactoring the code... as well as ignoring the linter warning